### DISCUSSION TITLE
Basic UI/image display for people with some usable vision or cited people wanting to help the blind

### DISCUSSION THREAD

**Discussion by alexoloopios** [samtupy/nvgt#136] [Ideas]
I know this might sound very weird to some people, but my girlfriend Chloe and I were talking about this, and Sam even thought it an interesting idea, that we work on a way of either having some basic user interface thing or away for an image or different images to be displayed Depending on what part of a program the user is on.
An example of this could be STAR, we could have an image that has representations of what would usually be actual UI elements, so then people with some usable vision who might not be used to audio only interfaces, or even cited people that want to be able to use the tools we do, especially people who are making TTS but don't want to deal with doing that line by line crap, can have an easier time with the program. Yeah we kind of crossed paths between NVG T and Star but that's the only example I could think of off the top of my head.
We have obviously got more important things to prioritise, but I just wanted to put this out there as we could definitely do this at some point in the future.

**Comment by harrymkt** [comment]
Wouldn't this against its name? The Nonvisual Game Toolkit?

**Reply by alexoloopios** [reply]
Well sort of, but there are actually some people out there that have enough vision to get away with large print interfaces and stuff like that, we want to be as inclusive as possible and there may be some people out there that are not used to this sort of thing so would be nice to do it for that reason

**Comment by ethindp** [comment]
The problem with this idea is the implementation. Chloe would be the only one who could actually validate our implementation, and the coordinates right so they display correctly on all screens would be an utter nightmare. I have zero idea how mainstream software do it. We may not need to call into APIs like Vulkan or OpenGL, but it would still be very painful to actually do. I can do "display text at the center of the screen" but definitely can't do "display this menu".

We could display a single image, perhaps, in the main window, but anything more than that and we risk needing to implement 3D rendering and all that, which would get very painful very fast.

**Reply by Jonathans859** [reply]
I could do it as well,  have enough vision left for the basic things. But yeah get it. It would be damn cool to have things like displaying a menu or form, but I see the issues.
